src/initialize.cc

```diff
@@ -57,13 +57,18 @@ class LibraryInitializer {
         Engine::Get()->Start();
       },
       []() {
-        // Make children single threaded since they are typically workers
-        dmlc::SetEnv("MXNET_CPU_WORKER_NTHREADS", 1);
-        dmlc::SetEnv("OMP_NUM_THREADS", 1);
+        // Conservative thread management for multiprocess workers
+        const size_t mp_worker_threads = dmlc::GetEnv("MXNET_MP_WORKER_NTHREADS", 1);
+        const size_t mp_omp_threads = dmlc::GetEnv("MXNET_MP_OMP_NUM_THREADS", 1);
```

Comment:
libgomp doesnt play well with fork: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52303 . Which means user using libgomp and increasing the num threads can run into a deadlock.

Reply:
I see, do you suggest completely turn off omp in worker process regardless?